### ISSUE TITLE
[YSQL][Profiles] Yugabyte login attempt counter reset to 0 after giving empty password

### ISSUE THREAD

**Issue by rjalan-yb** [yugabyte/yugabyte-db#15435]
Jira Link: [DB-4551](https://yugabyte.atlassian.net/browse/DB-4551)
### Description

When we try login to one node as yugabyte through any other node or machine with different hostname, the counter of failed attempts get reset to 0 only for Yugabyte role. Not for postgres or any other user created roles. Can you please take a look?

Steps to reproduce the issue:
1. Create at least a 2 node cluster. Lets say one node has IP1 and other has IP2.
2. Create a profile with 3 attempts and attach the Yugabyte role to profile.
3. Try some failure login attempts to Yugabyte user by giving wrong password, use Node2 to connect to Node 1.
`IP2 >> ./ysqlsh -h <IP1> -U yugabyte;`
4. Validate that failed attempt should increase to 2 after 2 failures.
```
yugabyte=# select * from pg_yb_role_profile;
 rolprfrole | rolprfprofile | rolprfstatus | rolprffailedloginattempts | rolprflockeduntil
------------+---------------+--------------+---------------------------+-------------------
      13287 |         16384 | o            |                         2 |
(1 row)
```

5.  Try login now by giving an empty password.
`IP2 >> ./ysqlsh -h <IP1> -U yugabyte;`
6. Failed attempts would have gone to 0.
```
yugabyte=# select * from pg_yb_role_profile;
 rolprfrole | rolprfprofile | rolprfstatus | rolprffailedloginattempts | rolprflockeduntil
------------+---------------+--------------+---------------------------+-------------------
      13287 |         16384 | o            |                         0 |
(1 row)
```

This does not happen with any other user like postgres or any other user defined roles. But only with postgres user.

Issue 2 -> Yugabyte Role Login Failure Attempt count resets automatically to 0 after some time.
Issue 3 -> Yugabyte role login attempt counter reset to 0 when logged in with another superuser

**Comment by vrajat** [comment]
I am not able to reproduce the reset to 0. I tried with and without c6c087db1f8bc6ee2d5b1c7f7ea45e6338824965. 

I setup a 2-node cluster using yugabyted and followed the steps to reproduce. I tried login attempts to `yugabyte` to the same host and the other host by not supplying a password. One observation is that the `rolprffailedloginattempts` is NOT incremented. However, it was not set to zero.

I also repeated the steps with another super user with and without the above commit. 

@rjalan-yb was the cluster a multi-region or multi-zone one?

I'll need your help in reproducing the issue.

**Comment by rjalan-yb** [comment]
Hey @vrajat, this is a multi-zone setup. Can you also try from your local machine to host?

**Comment by vrajat** [comment]
I am able to reproduce the behaviour in portal. My theory is that for some reason, the auth code is returning `STATUS_OK` for `yugabyte` user and counter is reset to zero in this [code location](https://github.com/yugabyte/yugabyte-db/blob/master/src/postgres/src/backend/libpq/auth.c#L665).

**Comment by vrajat** [comment]
My current suspicion is that there is process logging into the database using local all yugabyte trust. Circumstantial evidence are:
* I commented out the above line in ysql_hba.conf and restarted the process. This line was added back.
* The counter does not take on random numbers or increment/decrement by different values. The only issue is that it is reset to 0.
* If yugabyte account is locked, then the counter is not reset to 0.

I’ll run the following experiments:
1. Try to reproduce the reset to 0 for yugabyte on a 1 node cluster. This will give me more confidence that it is not a caching issue.
2. Add another condition that auth type should not be uATrust at https://github.com/yugabyte/yugabyte-db/blob/master/src/postgres/src/backend/libpq/auth.c#L665. A release with this change should stop the reset to 0.
3. Add a long sleep and check which process is using uATrust auth method if any including stack trace.
4. Modify hba_conf using tserver logs and verify that local all yugabyte trust gets added to hba_conf. Gives confidence that YBA requires this auth method for maintenance tasks.
